python/ray/util/sgd/v2/trainer.py

```diff
@@ -244,8 +371,56 @@ def to_tune_trainable(
         Returns:
             A Trainable that can directly be passed into ``tune.run()``.
         """
+        if not TUNE_INSTALLED:
+            raise ValueError("Tune is not installed. Please install ray["
+                             "tune] to use the Tune integration.")
 
-        def trainable_func(config: Dict[str, Any]) -> T:
-            pass
+        return _create_tune_trainable(train_func, self._backend,
+                                      self._num_workers, self._use_gpu,
+                                      self._resources_per_worker)
 
-        raise NotImplementedError
+
+def _create_tune_trainable(train_func, backend, num_workers, use_gpu,
+                           resources_per_worker):
+    """Creates a Tune Trainable class for SGD training.
+
+    This function populates class attributes and methods.
+    """
+
+    # TODO(amog): Implement checkpointing for Tune.
+    def tune_function(config, checkpoint_dir=None):
+        trainer = Trainer(
+            backend=backend,
+            num_workers=num_workers,
+            use_gpu=use_gpu,
+            resources_per_worker=resources_per_worker)
+
+        trainer.start()
+
+        iterator = trainer.run_iterator(train_func, config)
+
+        for results in iterator:
+            first_worker_results = results[0]
+
+            tune.report(**first_worker_results)
+
+    trainable_cls = wrap_function(tune_function)
```

Comment:
Do we have to call `trainer.shutdown()` anywhere?

Reply:
Yes, good point. I added this at the end.